core/templates/components/state-editor/state-hints-editor/state-hints-editor.component.ts

```diff
@@ -13,231 +13,217 @@
-      };
-
-      $scope.deleteHint = function(index, evt) {
-        // Prevent clicking on the delete button from also toggling the
-        // display state of the hint.
-        evt.stopPropagation();
-
-        AlertsService.clearWarnings();
-        NgbModal.open(DeleteHintModalComponent, {
-          backdrop: true,
-        }).result.then(function() {
-          if (StateSolutionService.savedMemento &&
-            StateHintsService.savedMemento.length === 1) {
-            openDeleteLastHintModal();
-          } else {
-            StateHintsService.displayed.splice(index, 1);
-            StateHintsService.saveDisplayedValue();
-            ctrl.onSaveHints(StateHintsService.displayed);
-          }
-
-          if (index === StateHintsService.getActiveHintIndex()) {
-            StateHintsService.setActiveHintIndex(null);
-          }
-        }, function() {
-          AlertsService.clearWarnings();
-        });
-      };
-
-      $scope.onSaveInlineHint = function() {
-        StateHintsService.saveDisplayedValue();
-        ctrl.onSaveHints(StateHintsService.displayed);
-      };
-
-      $scope.toggleHintCard = function() {
-        $scope.hintCardIsShown = !$scope.hintCardIsShown;
-      };
-
-      ctrl.$onInit = function() {
-        $scope.EditabilityService = EditabilityService;
-        $scope.StateHintsService = StateHintsService;
-        $scope.hintCardIsShown = (
-          !WindowDimensionsService.isWindowNarrow());
-        StateHintsService.setActiveHintIndex(null);
-        $scope.canEdit = EditabilityService.isEditable();
-        $scope.getStaticImageUrl = function(imagePath) {
-          return UrlInterpolationService.getStaticImageUrl(imagePath);
-        };
-        // When the page is scrolled so that the top of the page is above
-        // the browser viewport, there are some bugs in the positioning of
-        // the helper. This is a bug in jQueryUI that has not been fixed
-        // yet. For more details, see http://stackoverflow.com/q/5791886
-        $scope.HINT_LIST_SORTABLE_OPTIONS = {
-          axis: 'y',
-          cursor: 'move',
-          handle: '.oppia-hint-sort-handle',
-          items: '.oppia-sortable-hint',
-          revert: 100,
-          tolerance: 'pointer',
-          start: function(e, ui) {
-            ExternalSaveService.onExternalSave.emit();
-            StateHintsService.setActiveHintIndex(null);
-            ui.placeholder.height(ui.item.height());
-          },
-          stop: function() {
-            StateHintsService.saveDisplayedValue();
-            ctrl.onSaveHints(StateHintsService.displayed);
-          }
-        };
-        StateEditorService.updateStateHintsEditorInitialised();
-      };
+  openAddHintModal(): void {
+    if (this.stateHintsService.displayed.length >= 5) {
+      return;
     }
-  ]
-});
+    this.alertsService.clearWarnings();
+    this.externalSaveService.onExternalSave.emit();
+
+    const addHintSuccess = (result: {hint: Hint}): void => {
```

Comment:
Should we define a new type for the result object and use that here?

Reply:
Yes